src/armor_layers.cpp

```diff
@@ -112,387 +106,499 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
-        i += fold_and_print( w_sort_middle, point( 0, i ), win_width, c_light_gray, message );
-    }
-}
 
 std::string clothing_layer( const item &worn_item )
 {
     std::string layer;
 
     if( worn_item.has_flag( "PERSONAL" ) ) {
-        layer = _( "This is in your personal aura." );
+        layer = _( "(p.aur)" );
     } else if( worn_item.has_flag( "SKINTIGHT" ) ) {
-        layer = _( "This is worn next to the skin." );
+        layer = _( "(skin)" );
     } else if( worn_item.has_flag( "WAIST" ) ) {
-        layer = _( "This is worn on or around your waist." );
+        layer = _( "(waist)" );
     } else if( worn_item.has_flag( "OUTER" ) ) {
-        layer = _( "This is worn over your other clothes." );
+        layer = _( "(outer)" );
     } else if( worn_item.has_flag( "BELTED" ) ) {
-        layer = _( "This is strapped onto you." );
+        layer = _( "(strap)" );
     } else if( worn_item.has_flag( "AURA" ) ) {
-        layer = _( "This is an aura around you." );
+        layer = _( "(aura)" );
     }
 
     return layer;
 }
-
-std::vector<std::string> clothing_properties(
-    const item &worn_item, const int width, const Character &c )
-{
-    std::vector<std::string> props;
-    props.reserve( 5 );
-
-    const std::string space = "  ";
-    props.push_back( string_format( "<color_c_green>[%s]</color>", _( "Properties" ) ) );
-    props.push_back( name_and_value( space + _( "Coverage:" ),
-                                     string_format( "%3d", worn_item.get_coverage() ), width ) );
-    props.push_back( name_and_value( space + _( "Encumbrance:" ),
-                                     string_format( "%3d", worn_item.get_encumber( c ) ),
-                                     width ) );
-    props.push_back( name_and_value( space + _( "Warmth:" ),
-                                     string_format( "%3d", worn_item.get_warmth() ), width ) );
-    props.push_back( name_and_value( space + string_format( _( "Storage (%s):" ), volume_units_abbr() ),
-                                     format_volume( worn_item.get_storage() ), width ) );
-    return props;
-}
-
-std::vector<std::string> clothing_protection( const item &worn_item, const int width )
+enum sort_armor_columns : int {
+    column_cond,
+    column_encumb,
+    column_storage,
+    column_warmth,
+    column_divider,
+    column_coverage,
+    column_bash,
+    column_cut,
+    column_acid,
+    column_fire,
+    column_env,
+    column_num_entries
+};
+struct sort_armor_col_data {
+    private:
+        int width;
+        int cut_off;
+        std::string name;
+        std::string description;
+    public:
+        sort_armor_col_data( int width, int cut_off, std::string name, std::string description ):
+            width( width ), cut_off( cut_off ), name( name ), description( description ) {
+        }
+        int get_width() {
+            return width;
+        }
+        std::string get_name() {
+            return ( _( name ) ).substr( 0, cut_off );
```

Comment:
Please use `utf8_truncate` instead of `substr`. The later will cut right through UTF-8 characters that consist of multiple bytes and thereby produce the wrong output.

Reply:
@akozhevn I see space is tight in small resolution, but this is not really a good way to solve the alignment problem: there's even not a translation comment to tell the translator to limit the character count in their translation, so even if the string is truncated using `utf8_truncate` it will still likely produce some nonsense in other languages.

A solution I can current think of is to split the columns into pages if space is tight (i.e. if the added width of all translated column headers, including padding, exceeds screen width), and allow scrolling the pages using tab and backtab. This way we can fully utilize the screen space, need no heavy abbreviation, and ensure that the data can be viewed in any resolution.

> if data size is same and columns sizes are different it'll be harder to read
You can always take the largest header width as the width for every column.